src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> | object {
+  if (items.length === 0) {
+    return {};
+  }
+
+  const result: GroupsMap<T> = {};
+
+  for (const item of items) {
+    const groupKey = item[key] as string;
```

Comment:
Casting `item[key]` to string assumes that the key property of T is always stringifiable. This may not always be the case. It's better to enforce this constraint in the function's type parameters by requiring that `key` is a keyof T where T[key] extends a type that can be stringified.